pkg/services/publicdashboards/service/service_test.go

```diff
@@ -1295,35 +1295,57 @@ func assertOldValueIfNull(t *testing.T, expectedValue bool, oldValue bool, nulla
 }
 
 func TestDeletePublicDashboard(t *testing.T) {
+	store := NewFakePublicDashboardStore(t)
+	pubdash := &PublicDashboard{Uid: "2", OrgId: 1, DashboardUid: "uid"}
+	type mockResponse struct {
+		PublicDashboard *PublicDashboard
+		Err             error
+	}
 	testCases := []struct {
 		Name             string
 		AffectedRowsResp int64
 		ExpectedErrResp  error
 		StoreRespErr     error
+		dashboard        *dashboards.Dashboard
+		mockStore        *mockResponse
 	}{
 		{
-			Name:             "Successfully deletes a public dashboards",
+			Name:             "Successfully deletes a public dashboard",
 			AffectedRowsResp: 1,
 			ExpectedErrResp:  nil,
 			StoreRespErr:     nil,
+			dashboard:        &dashboards.Dashboard{UID: "uid", OrgID: 1, IsFolder: false},
+			mockStore:        &mockResponse{pubdash, nil},
 		},
 		{
-			Name:             "Public dashboard not found",
+			Name:             "Public dashboard not found by UID",
 			AffectedRowsResp: 0,
-			ExpectedErrResp:  nil,
+			ExpectedErrResp:  ErrPublicDashboardNotFound,
 			StoreRespErr:     nil,
+			dashboard:        &dashboards.Dashboard{UID: "uid", OrgID: 1, IsFolder: false},
+			mockStore:        &mockResponse{pubdash, ErrDashboardNotFound},
```

Comment:
The store would return `nil, nil` when there is no pubdash with the uid provided.